tests/integration/extenders/NotificationTest.php

```diff
@@ -37,6 +48,21 @@ public function notification_type_exists_if_added()
 
         $this->assertArrayHasKey('customNotificationType', Notification::getSubjectModels());
     }
+
+    /**
+     * @test
+     */
+    public function notification_driver_exists_if_added()
+    {
+        $this->extend((new Extend\Notification())->driver(
+            'customNotificationDriver',
+            CustomNotificationDriver::class
+        ));
+
+        $this->app();
+
+        $this->assertArrayHasKey('customNotificationDriver', NotificationSyncer::getNotificationDrivers());
+    }
```

Comment:
Any way we can test that we can add enabled types via a driver?

Reply:
That's what I was trying to do, but I'm not sure yet, we need a way to access the enabled driver names for specific types, and that's only in the user preferences array which is protected.

Reply:
Since the code works by extending the `'flarum.notification.blueprints'` singleton, couldn't we resolve that and check its contents?

Reply:
Ah.. I did not know we had access to the container here.